src/ucs/datastruct/list.c

```diff
@@ -0,0 +1,78 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "list.h"
+
+#include <ucs/debug/log.h>
+#include <ucs/debug/memtrack_int.h>
+#include <ucs/sys/math.h>
+
+void ucs_list_print_links(ucs_list_link_t *head, unsigned int num_nodes)
+{
+    ucs_list_link_t *node;
+    unsigned int i;
+
+    ucs_debug("List head: %p prev=%p next=%p\n", (void *)head, (void *)head->prev,
+              (void *)head->next);
+
+    node = head->next;
+    for (i = 0; i < num_nodes; i++) {
+        ucs_debug("Node #%u: %p prev=%p next=%p\n", i, (void *)node,
+                  (void *)node->prev, (void *)node->next);
+        node = node->next;
+    }
+}
+
+void ucs_list_shuffle_order(ucs_list_link_t *head, unsigned int num_nodes)
+{
+    ucs_list_link_t **nodes_array;
+    ucs_list_link_t *temp_node;
+    int i, j;
+
+    if (num_nodes <= 1) {
+        ucs_debug("Nothing to shuffle, num_nodes=%u", num_nodes);
+        return;
+    }
+
+    nodes_array = (ucs_list_link_t **)ucs_malloc(
+        num_nodes * sizeof(*nodes_array), "nodes_array");
```

Comment:
2nd arg name - give more context (see usage examples)

Reply:
Renamed `nodes_array` to `node_pointers` it highlights that the array holds pointers to list nodes, but I don't mind renaming again if you have another suggestion